### PR TITLE
feat(datepicker): add a drop-in datepicker input

### PR DESCRIPTION
**Pull request by nadiia**

### CODE REVIEW

File: src/datepicker/datepicker.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+  handleKeyDown = (event: KeyboardEvent) => {
+    if (!this.state.isOpen && event.keyCode === 40) {
+      this.open();
+    }
+  };
+
+  render() {
+    const {overrides = {}} = this.props;
+    const [InputC, inputProps] = getOverrides(overrides.Input, Input);
```
Comment (tajo) [inline src/datepicker/datepicker.js]: What the `C` stands for? :)

**Comment by gergelyke** [comment]
Once merged, should we say that datepicker is not unstable anymore?

One potential issue I ran into: when the datepicker is opened in the popover, when I click outside of it the Datepicker is not closed - is that on purpose?

**Comment by tajo** [comment]
Keyboard control is now global and changes the state of other instances too.

shortcuts_are_global

**Comment by tajo** [comment]
When you try to open the year / month selector, the whole page jumps up.

jump

**Comment by tajo** [comment]
**Nitpick:** When you click fast (double-click) on arrows, it selects the text (month or day names). `user-select: none` would fix that. 

select

**Comment by tajo** [comment]
Anyway, thanks for working on this! It's really complex and overall it works very well!

File: src/datepicker/datepicker.js (hunk 1 above)
Comment (nadiia) [reply src/datepicker/datepicker.js]: `Component`,  didn't want to rename the top level imports 🤷‍♀️

Comment (gergelyke) [reply src/datepicker/datepicker.js]: should we be explicit, and name it `InputComponent`?

**Comment by tajo** [comment]
**It looks and works great!** All the bugs are fixed and the keyboard control is pretty intuitive to me now!

One one last tiny thing: The page jumps a bit when you first hit `Arrow Down` after the calendar is open.

jump

**Pull request merged**